### DISCUSSION TITLE
UBL Not Working on 2.0.2i

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1862] [Q&A]
Hello

After many bad prints and attempts to find solutions, it appears that UBL is not working on my Ender 3 V2 with version 2.0.2i and possibly the previous version too.

All the issues I was having disappeared once I reverted back to 1.3.5b (E3V2-UBL-BLTouch-15x15-HS-v4.2.2-v1.3.5.b).

I have used UBL for about a year before and it worked flawlessly and below is my starting Gcode:

```
G28 ; Home all axes
G29 L0 ; Load default mesh from EEPROM.
G29 A ; Activate UBL.
G29 J ; Get the bed tilt in 3-points.
```

Below are the firmware changes that I applied to 2.02i before compiling:

![Untitled](https://user-images.githubusercontent.com/65966061/155155481-2e4d78c8-ff7b-4b79-9106-fc0bafe8eb06.png)

The issue appeared in the first layer where in some areas of the bed, the nozzle is printing too close to the surface resulting in terrible quality or the layer coming off. I remember I was getting decent prints when I did not have a Bltouch or used any ABL so It seems that the UBL mesh is not applied or the math is not correct. And yes, I have leveled the bed and calibrated the Z offset and created brand new mesh few million times in the past days until I determined (pretty late) that the firmware is the issue after eliminating the possibility of hardware issue.

I hope this gets fixed in 2.1. I'm gonna stick with 1.3.5b for now.

**Comment by micah-simmler** [comment]
> Hello
> 
> After many bad prints and attempts to find solutions, it appears that UBL is not working on my Ender 3 V2 with version 2.0.2i and possibly the previous version too.
> 
> All the issues I was having disappeared once I reverted back to 1.3.5b (E3V2-UBL-BLTouch-15x15-HS-v4.2.2-v1.3.5.b).
> 
> I have used UBL for about a year before and it worked flawlessly and below is my starting Gcode:
> 
> ```
> G28 ; Home all axes
> G29 L0 ; Load default mesh from EEPROM.
> G29 A ; Activate UBL.
> G29 J ; Get the bed tilt in 3-points.
> ```
> 
> Below are the firmware changes that I applied to 2.02i before compiling:
> 
> ![Untitled](https://user-images.githubusercontent.com/65966061/155155481-2e4d78c8-ff7b-4b79-9106-fc0bafe8eb06.png)
> 
> The issue appeared in the first layer where in some areas of the bed, the nozzle is printing too close to the surface resulting in terrible quality or the layer coming off. I remember I was getting decent prints when I did not have a Bltouch or used any ABL so It seems that the UBL mesh is not applied or the math is not correct. And yes, I have leveled the bed and calibrated the Z offset and created brand new mesh few million times in the past days until I determined (pretty late) that the firmware is the issue after eliminating the possibility of hardware issue.
> 
> I hope this gets fixed in 2.1. I'm gonna stick with 1.3.5b for now.

I have the same issue I am about to try another print but I think I have it working better changing the gcode to G29 J2 
I am on v2.0.1

**Reply by BlueWings172** [reply]
@micah-simmler 
I have tried J2 before and it did not solve the issue.
